config/deploy.rb

```diff
@@ -37,6 +37,17 @@ def main_deploy_server
 
 set :local_user, ENV["USER"]
 
+set :fnm_path, "$HOME/.fnm"
+set :fnm_install_command, "curl -fsSL https://fnm.vercel.app/install | " \
+                          "bash -s -- --install-dir \"#{fetch(:fnm_path)}\""
+set :fnm_update_command, "#{fetch(:fnm_install_command)} --skip-shell"
+set :fnm_setup_command, -> do
```

Comment:
> I agree! No idea what's going on here 🤔. Maybe next week we can share a screen and try to find out.

Hi @javierm, 

Debugging a few deployments, I found that when executing the `fnm_setup_command`, it fails when `fnm` is not installed as expected, but it does not throw an exception, so the `fnm_install_command` is never reached. This silent failure is because the command `fnm_setup_command` uses eval, and the failure happens within the bash eval function.

Adding any `fnm` command in the `fnm_setup_command` fails as expected and throws the exception. I tried the following, and it worked fine:

```
set :fnm_setup_command, -> do
                          "export PATH=\"#{fetch(:fnm_path)}:$PATH\" && " \
                            "cd #{release_path} && fnm env && eval \"$(fnm env)\""
                        end
```